src/services/CookieAuth.php

```diff
@@ -48,16 +55,37 @@ public function tryAuth(): AuthResponse
         }
         $res = $req->fetch();
         $userid = (int) $res['userid'];
+
         // when doing auth with cookie, we take the token_team value
         // make sure user is in team because we can't trust it
         $TeamsHelper = new TeamsHelper($this->tokenTeam);
         if (!$TeamsHelper->isUserInTeam($userid)) {
             throw new UnauthorizedException();
         }
 
+        $this->enforceMfa($res);
+
         $this->AuthResponse->userid = $userid;
         $this->AuthResponse->mfaSecret = $res['mfa_secret'];
         $this->AuthResponse->selectedTeam = $this->tokenTeam;
         return $this->AuthResponse;
     }
+
+    /**
+     * Force user to login again to activate MFA if it is enforced for local auth and there is no mfaSecret
+     */
+    private function enforceMfa(array $res): void
```

Comment:
shouldn't this logic be extracted from CookieAuth and be in a more general "local auth" file? Seems like it is not cookie specific. Maybe in LocalAuth?